src/Email/Address.php

```diff
@@ -0,0 +1,116 @@
+<?php
+
+namespace Kirby\Email;
+
+use Kirby\Cms\User;
+use Kirby\Cms\Users;
+use Kirby\Exception\InvalidArgumentException;
+use Kirby\Toolkit\A;
+use Kirby\Toolkit\V;
+
+/**
+ * An email address with optional name
+ *
+ * @package   Kirby Email
+ * @author    Nico Hoffmann <nico@getkirby.com>
+ * @link      https://getkirby.com
+ * @copyright Bastian Allgeier
+ * @license   https://opensource.org/licenses/MIT
+ * @since     5.0.0
+ */
+class Address
+{
+	public function __construct(
+		protected string $email,
+		public string|null $name = null
+	) {
+		if (V::email($email) === false) {
+			throw new InvalidArgumentException(sprintf('"%s" is not a valid email address', $email));
+		}
+	}
+
+	/**
+	 * Creates one or multiple address objects from a
+	 * User object, simple string or email-name key-value pairs
+	 *
+	 * @throws \Kirby\Exception\InvalidArgumentException email address is invalid
+	 */
+	public static function factory(
+		Users|array|User|string $emails,
+		bool $multiple = false
+	): static|array {
+		// ensure we can iterate over $emails
+		if (is_iterable($emails) === false) {
+			$emails = A::wrap($emails);
+		}
+
+		$addresses = [];
+
+		foreach ($emails as $address => $name) {
+			$addresses[] = match (true) {
+				$name instanceof User => new static(
+					email: $name->email(),
+					name: $name->name()
+				),
+
+				is_string($address) => new static(
+					email: $address,
+					name: $name
+				),
+
+				default => new static(email: $name)
+			};
+		}
+
+		if ($multiple === false) {
+			return $addresses[0];
+		}
+
+		return $addresses;
+	}
+
+	/**
+	 * Returns the email address
+	 */
+	public function email(): string
+	{
+		return $this->email;
+	}
+
+	/**
+	 * Returns the name, if available
+	 */
+	public function name(): string|null
+	{
+		return $this->name;
+	}
+
+	/**
+	 * Returns one or multiple addresses as array
+	 * where the emails are the keys and the names the values
+	 */
+	public static function resolve(array|Address $address): array
```

Comment:
I think array usage that doesn't includes `Address` object is invalid:

````php
// single
Address::resolve(['homer@simpson.com']);

// multiple
Address::resolve([
  'homer@simpson.com' => null,
  'lisa@simpson.com'  => 'Lisa Simpson'
]);
````

If array items are should to be `Address` objects, it would be nice to have a type check.

Reply:
I'll make the method description more explicit. It is only meant to resolve Address objects and arrays of Address objects. There should be no need to support other formats.